src/arm7tdmi.rs

```diff
@@ -86,12 +92,45 @@ impl Arm7tdmi {
     fn branch_link(&mut self, op_code: u32) {
         todo!("Branch Link")
     }
+
+    fn mov(&mut self, op_code: u32) {
+        // bits [24-21] are the RD
+        let rd = (op_code & 0x00_00_F0_00) >> 12;
+        println!("RD: {:?}", rd);
+
+        // 25th bit is I = Immediate Flag
+        let immediate: bool = (op_code & 0x02_00_00_00) >> 25 == 1;
+        println!("Immediate: {:?}", immediate);
+
+        // 20th bit is S = Condition Set
+        if op_code & 0x00_08_00_00 > 0 {
+            todo!("Condition set")
+        }
+
+        if immediate {
+            // bits [7-0] are the immediate value
+            let immediate_value = op_code & 0x00_00_00_FF;
+            println!("value: {:?}", immediate_value);
+
+            // the instruction is MOV RD, immediate_value
+            self.registers[rd as usize] = immediate_value;
+        } else {
+            todo!("Not implemented yet.");
+        }
+
+        self.program_counter += 4;
+    }
 }
 
 #[derive(Debug, PartialEq, Eq)]
 pub(crate) enum ArmModeInstruction {
     Branch = 0x0A_00_00_00,
     BranchLink = 0x0B_00_00_00,
+
+    /// 27-26 must be 0b00
+    /// 24-21 must be 0b1101
+    /// 19-16 must be 0b0000
+    Mov = 0x01_A0_00_00,
```

Comment:
As the Data Processing instructions are very similar it can be refactor in a future